Structure_Engine/Compute/Integrate.cs

```diff
@@ -82,11 +83,46 @@ public static partial class Compute
                 results["Asy"] = 0;
                 results["Asz"] = 0;
 
-                #endregion 
+                #endregion
                 return results;
             }
+            else if (curves.Count == 0)
```

Comment:
Not sure I like the copy paste here..

Understand it has to do with wanting to give an error or not, but maybe worth changing to something like adding a private "ZeroConstantsDictionary" method (or similar better name if you can think of anything) that returns a dictionary filled up with the 0 values, and then return that method instead.